test/filter-test.js

```diff
@@ -0,0 +1,263 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const FullNode = require('../lib/node/fullnode');
+const {filters} = require('../lib/blockstore/common');
+const {BufferSet} = require('buffer-map');
+const {opcodes} = require('../lib/script/common');
+const FilterIndexer = require('../lib/indexer/filterindexer');
+const Block = require('../lib/primitives/block');
+const Golomb = require('../lib/golomb/golomb');
+const {U64} = require('n64');
+const BasicFilter = require('../lib/golomb/basicFilter');
+
+class TestFilter extends Golomb {
+  constructor() {
+    super(21, new U64(744931));
+  }
+}
+
+class TestBlock extends Block {
+  constructor() {
+    super();
+  }
+
+  toTestFilter(view) {
+    const hash = this.hash();
+    const key = hash.slice(0, 16);
+    const items = new BufferSet();
+
+    for (let i = 0; i < this.txs.length; i++) {
+    const tx = this.txs[i];
+
+    for (const output of tx.outputs) {
+        if (output.script.length === 0)
+        continue;
+
+        // In order to allow the filters to later be committed
+        // to within an OP_RETURN output, we ignore all
+        // OP_RETURNs to avoid a circular dependency.
+        if (output.script.raw[0] === opcodes.OP_RETURN)
+        continue;
+
+        items.add(output.script.raw);
+    }
+    }
+
+    for (const [, coins] of view.map) {
+    for (const [, coin] of coins.outputs) {
+        if (coin.output.script.length === 0)
+        continue;
+
+        items.add(coin.output.script.raw);
+    }
+    }
+
+    return new TestFilter().fromItems(key, items);
+  }
+
+  toFilter(view, filterType) {
+    switch (filterType) {
+        case filters.BASIC:
+            return this.toBasicFilter(view);
+        case filters.TEST:
+            return this.toTestFilter(view);
+        default:
+            return null;
+    }
+  }
+}
+
+Block.fromRaw = (data, enc) => {
+    if (typeof data === 'string')
+        data = Buffer.from(data, enc);
+    return new TestBlock().fromRaw(data);
+};
+
+const node = new FullNode({
+  network: 'regtest',
+  memory: true,
+  indexFilter: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+filters['TEST'] = 0xffffffff;
+
+describe('Filter', function() {
+  this.timeout(15000);
+
+  before(async () => {
+      const testFilterIndex = new FilterIndexer({
+                                  network: node.network,
+                                  logger: node.logger,
+                                  blocks: node.blocks,
+                                  chain: node.chain,
+                                  memory: node.config.bool('memory'),
+                                  prefix: node.config.str('index-prefix', node.config.prefix),
+                                  filterType: 'TEST'
+                              });
+      node.filterIndexers.set('TEST', testFilterIndex);
+
+      await node.open();
+  });
```

Comment:
```suggestion
  before(async () => {
    const testFilterIndex = new FilterIndexer({
      network: node.network,
      logger: node.logger,
      blocks: node.blocks,
      chain: node.chain,
      memory: node.config.bool('memory'),
      prefix: node.config.str('index-prefix', node.config.prefix),
      filterType: 'TEST'
    });
    node.filterIndexers.set('TEST', testFilterIndex);

    await node.open();
  });
```

Reply:
theres more indentation nits like this below, just try to keep everything at 2 spaces.